keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -382,10 +388,107 @@ class ClusterHandler(
     return spec
   }
 
-  private fun List<ResourceDiff<ServerGroup>>.createsNewServerGroups() =
-    any {
-      !it.isCapacityOrAutoScalingOnly()
+  override suspend fun actuationInProgress(resource: Resource<ClusterSpec>) =
+    resource
+      .spec
+      .locations
+      .regions
+      .map { it.name }
+      .any { region ->
+        orcaService
+          .getCorrelatedExecutions("${resource.id}:$region")
+          .isNotEmpty()
+      }
+
+  /**
+   * Retrieve server group entity tags from clouddriver to identity the "source" of the server group (pipeline or
+   * orchestration), then retrieve execution details from orca to determine the deployment strategy used for that
+   * server group.
+   */
+  private suspend fun ServerGroup.discoverDeploymentStrategy(): ClusterDeployStrategy? {
+    return coroutineScope {
+      val entityTags = async {
```

Comment:
Note that IntelliJ Is flagging the two `async` blocks in this method with the warning: _Redundant 'async' call may be reduced to 'kotlinx.coroutines.withContext'_

Reply:
Yeah, it did for me too, but I thought that form was much less readable/obvious than `async`.